spec/spec.md

```diff
@@ -80,21 +80,24 @@ DELETE <ECHO_SERVER_URL>/clients/<CLIENT_ID>
 
 ### Send Notification
 
-> **Note**
-> This will have E2E encryption in the future but for now is designed without encryption.
-
 ```
 POST <ECHO_SERVER_URL>/clients/<CLIENT_ID>
 {
     "id": "0000-0000-0000-0000"
-    "payload": {
-        "title": "New Request",
-        "description": "You have a signing request."
-    }
+    "payload": "<topic>:<flags>:<blob>"
 }
 ```
 
 - `CLIENT_ID`: This is used to map to the device token
 - `id`: This is generated by the Relay to prevent duplicate notifications
-- `payload`: A JSON object that holds the relevant payload data. **NOTE:** This will become a string that is encrypted
-  by the dApp and not decrypted until received by the client
\ No newline at end of file
+- `payload`: A string with 3 values encoded and split with a `:` as a delimiter.
+  - `topic`: This is used by the SDKs to decrypt encrypted payloads on the client side
+  - `flags`: an integer value that uses bitmasks to encode information about the payload for the SDKs, the flags are below
+    ```
+    Encrypted  = 1 << 0
+    Sign       = 1 << 1
+    Auth       = 1 << 2
+    Chat       = 1 << 3
+    Push       = 1 << 4
+    ```
```

Comment:
don't get this part. Could it be just a binary value? 0 - unencrypted, 1 - encrypted ?

Reply:
It could but do we not want to be able to tell the SDKs apart?

Reply:
From client perspective, I don’t think it’s necessary

Reply:
Wallets wanted a way to distinguish them, cc @arein, I believe rainbow wanted to be able to tell if they were getting them from us or from potentially a third-party

Reply:
note that for chat you can have two types of notifications: 
chat_invite - unencrypted
chat_message - encrypted

how echo server decides if it should set the mutable_content flag (for iOS) if it only knows the api(Chat)?
correct me if I am wrong but cannot see other param.

but so far cannot find a usecase for an end client to need sdk - all encrypted will go to we3inbox

Reply:
It can set 2 flags at once, it uses bit masking so the request can be encrypted and chat, just chat, encrypted and push, just push, etc

Reply:
unencrypted will just appear on the screen

Reply:
Yes

Reply:
you can decide, I don't need to know the sdk right now but no one knows if we need it in the future

Reply:
I think we should keep it for now as it gives more information to Echo Server